src/app/routes/cpsAsset/getInitialData/processUnavailableMedia/index.js

```diff
@@ -0,0 +1,35 @@
+import { path, pathOr, assocPath } from 'ramda/src';
+
+const UNAVAILABLE_MEDIA_TEXT = 'unavailableMedia';
+const unavailableMediaBlock = {
+  type: UNAVAILABLE_MEDIA_TEXT,
+  model: {},
+  id: UNAVAILABLE_MEDIA_TEXT,
+};
+
+const addUnavailableMediaBlock = pageData => {
+  const blocks = pathOr([], ['content', 'model', 'blocks'], pageData);
+  const filteredBlocks = blocks.filter(block => block.type !== 'external_vpid');
+  return assocPath(
+    ['content', 'model', 'blocks'],
+    [unavailableMediaBlock, ...filteredBlocks],
+    pageData,
+  );
+};
+
+const transformer = pageData => {
+  const pageType = path(['metadata', 'type'], pageData);
+  if (pageType !== 'MAP') return pageData;
```

Comment:
Would be nice to have this type of logic up a level if it's not too much hassle -  ie, having it in `getInitialData`

It would make it clearer from looking at that file which preprocessors are actually applied to each page type

Reply:
Yeah, I considered this. I wanted to get it working at the lower level first.